Lib/idlelib/debugger.py

```diff
@@ -130,80 +154,187 @@ def make_gui(self):
-        self.bstep = b = Button(bframe, text="Step", command=self.step)
-        bl.append(b)
-        self.bnext = b = Button(bframe, text="Over", command=self.next)
-        bl.append(b)
-        self.bret = b = Button(bframe, text="Out", command=self.ret)
-        bl.append(b)
-        self.bret = b = Button(bframe, text="Quit", command=self.quit)
-        bl.append(b)
-        #
-        for b in bl:
-            b.configure(state="disabled")
-            b.pack(side="left")
-        #
-        self.cframe = cframe = Frame(bframe)
-        self.cframe.pack(side="left")
-        #
-        if not self.vstack:
-            self.__class__.vstack = BooleanVar(top)
-            self.vstack.set(1)
-        self.bstack = Checkbutton(cframe,
-            text="Stack", command=self.show_stack, variable=self.vstack)
-        self.bstack.grid(row=0, column=0)
-        if not self.vsource:
-            self.__class__.vsource = BooleanVar(top)
-        self.bsource = Checkbutton(cframe,
-            text="Source", command=self.show_source, variable=self.vsource)
-        self.bsource.grid(row=0, column=1)
-        if not self.vlocals:
-            self.__class__.vlocals = BooleanVar(top)
-            self.vlocals.set(1)
-        self.blocals = Checkbutton(cframe,
-            text="Locals", command=self.show_locals, variable=self.vlocals)
-        self.blocals.grid(row=1, column=0)
-        if not self.vglobals:
-            self.__class__.vglobals = BooleanVar(top)
-        self.bglobals = Checkbutton(cframe,
-            text="Globals", command=self.show_globals, variable=self.vglobals)
-        self.bglobals.grid(row=1, column=1)
-        #
-        self.status = Label(top, anchor="w")
-        self.status.pack(anchor="w")
-        self.error = Label(top, anchor="w")
-        self.error.pack(anchor="w", fill="x")
-        self.errorbg = self.error.cget("background")
-        #
-        self.fstack = Frame(top, height=1)
-        self.fstack.pack(expand=1, fill="both")
-        self.flocals = Frame(top)
-        self.flocals.pack(expand=1, fill="both")
-        self.fglobals = Frame(top, height=1)
-        self.fglobals.pack(expand=1, fill="both")
-        #
-        if self.vstack.get():
-            self.show_stack()
-        if self.vlocals.get():
-            self.show_locals()
-        if self.vglobals.get():
-            self.show_globals()
+
+        self.var_open_source_windows = BooleanVar(top, False)
+
+        self.pane = ttk.PanedWindow(self.top, orient='horizontal')
+        self.pane.grid(column=0, row=0, sticky='nwes')
+        self.top.grid_columnconfigure(0, weight=1)
+        self.top.grid_rowconfigure(0, weight=1)
+        self.left = left = ttk.Frame(self.pane, padding=5)
+        self.pane.add(left, weight=1)
+        controls = ttk.Frame(left)
+        col = 0
+        self.buttondata = {}
+        self.buttons = ['go', 'step', 'over', 'out', 'stop', 'prefs']
+        self.button_names = {'go':'Go', 'step':'Step', 'over':'Over',
```

Comment:
```suggestion
        self.button_names = {'go': 'Go', 'step': 'Step', 'over': 'Over',
```
IMO this looks neater

Reply:
For slices, no spaces is (usually) correct.  For annotations, space after is correct.  For displays, PEP 8 says nothing that I could find.   For no-space values, I think I prefer the Mark's original, so that the only space is between items.  I doubt the IDLE codebase is consistent.

Reply:
if no strong feelings, i'll keep the original